colearn_grpc/utils.py

```diff
@@ -91,19 +96,39 @@ async def iterator_to_weights_async(request_iterator, decode=True) -> Weights:
 
     weights_bytes = bytes(full_weights)
     if decode:
-        return decode_weights(weights_bytes)
+        return Weights(weights=decode_weights(weights_bytes), training_summary=training_summary)
     else:
         # On the client side we can't necessarily unpickle the Weights object because the relevant libraries might not
         # be importable. But we need to return a Weights object to match the MLI. So we wrap the pickled Weights in
         # another Weights object.
-        return Weights(weights=weights_bytes)
+        return Weights(weights=weights_bytes, training_summary=training_summary)
+
+
+def transform_training_summary_to_pb(ts: TrainingSummary) -> TrainingSummaryPb:
+    tsn = TrainingSummaryPb()
+    for field in ["dp_budget"]:
+        target = getattr(tsn, field)
+        source = getattr(ts, field)
+        for entry in source.__fields_set__:
```

Comment:
interesting way of setting protbuf structs, isn't supposed to have object getters and setter for all the fields?

Reply:
they have all kinds of helpers, but this is the easiest way of getting the fields.